Modules/TPC/src/TrendingTaskTPC.cxx

```diff
@@ -322,13 +281,73 @@ void TrendingTaskTPC::drawCanvasMO(TCanvas* thisCanvas, const std::string& var,
           // We try to convince ROOT to delete graphErrors together with the rest of the canvas.
           if (auto* pad = thisCanvas->GetPad(p + 1)) {
             if (auto* primitives = pad->GetListOfPrimitives()) {
-              primitives->Add(graphErrors); // TO-DO: Is this needed?
+              primitives->Add(graphErrors);
             }
           }
         }
       }
     }
-  } else if (trendType == "slices") {
+  } // Trending vs time
+  else if (trendType == "multigraphtime") {
+
+    TMultiGraph* multigraph = new TMultiGraph();
+    multigraph->SetName("MultiGraph");
+    TGraphErrors* Graphs[nuPa] = { NULL };
+
+    for (int p = 0; p < nuPa; p++) {
+      int iEntry = 0;
+      Graphs[p] = new TGraphErrors(nEntries);
+
+      while (myReader.Next()) {
+        const double timeStamp = (double)(*RetrieveTime);
+        const double dataPoint = (DataRetrieveVector->at(p)).RetrieveValue(typeName);
+        double errorX = 0.;
+        double errorY = 0.;
+
+        if (!err.empty()) {
+          errorX = (DataRetrieveVector->at(p)).RetrieveValue(errXName);
+          errorY = (DataRetrieveVector->at(p)).RetrieveValue(errYName);
+        }
+
+        Graphs[p]->SetPoint(iEntry, timeStamp, dataPoint);
+        Graphs[p]->SetPointError(iEntry, errorX, errorY); // Add Error to the last added point
+
+        std::string title = (DataRetrieveVector->at(p)).title;
+        const std::size_t posDivider = title.find("RangeX");
+        Graphs[p]->SetName(title.substr(posDivider, -1).data());
+        iEntry++;
+      }
+
+      myReader.Restart();
+      multigraph->Add(Graphs[p]);
+    } // for (int p = 0; p < nuPa; p++)
+
+    thisCanvas->cd(1);
+    multigraph->Draw("A pmc plc");
+    TLegend* legend = new TLegend(0., 0.1, 0.95, 0.9);
+    TIter nextgraph(multigraph->GetListOfGraphs());
+    legend->SetName("MultiGraphLegend");
+    legend->SetNColumns(2);
+    legend->SetTextSize(2.0);
+    TGraphErrors* gr;
+    TObject* obj;
+    while ((obj = nextgraph())) {
+      gr = (TGraphErrors*)obj;
+      legend->AddEntry(obj, gr->GetName(), "lpf");
+    }
+    // We try to convince ROOT to delete multigraph and legend together with the rest of the canvas.
+    if (auto* pad = thisCanvas->GetPad(1)) {
+      if (auto* primitives = pad->GetListOfPrimitives()) {
+        primitives->Add(multigraph);
+      }
+    }
+    if (auto* pad = thisCanvas->GetPad(2)) {
+      if (auto* primitives = pad->GetListOfPrimitives()) {
+        primitives->Add(legend);
+      }
+    }
```

Comment:
Can you try if this has the same effect:
```suggestion
  legend->Draw("same");
  multigraph->SetBit(TObject::kCanDelete);
  legend->SetBit(TObject::kCanDelete);
```

Reply:
If there are other places where objects are drawn on canvases, the pointer is lost after the function and the object should only be associated to the canvas you could also set the `kCanDelete` bit.

Reply:
This did unfortunately not work. I kept the previous version but made a dedicated function out of it to avoid code replication.